### PR TITLE
some clean up after zebra changes

### PR DESCRIPTION
**Pull request by tranma**
just a bit less savagery first

! @amosr @jystic

### CODE REVIEW

File: icicle-compiler/src/Icicle/Sea/Eval/Base.hs [hunk 1]
```diff
@@ -255,64 +257,67 @@ seaCompile' options cache input programs chords = do
...
-                     $ mergeBlocks (MergeOptions puller step' 100) pullid
+            ret <- runEitherT . joinErrors show show
+                 $ mergeBlocks (MergeOptions puller' step' 100) pullid
 
-                case ret of
-                  Left e ->
-                    Savage.error $ "error merge: " <> show e
-                  Right _ ->
-                    return ()
+            -- need to error straight from IO as piano is tied to IO
```
Comment (jystic) [inline icicle-compiler/src/Icicle/Sea/Eval/Base.hs]: not sure I understand this, why can't we use `EitherT`?

Comment (tranma) [inline icicle-compiler/src/Icicle/Sea/Eval/Base.hs]: `withPiano` is the continuation that will be given to `withCPiano :: ForeignPiano -> (CPiano -> IO a) -> IO a`

Comment (tranma) [inline icicle-compiler/src/Icicle/Sea/Eval/Base.hs]: maybe I could return the error rather than () though

Comment (jystic) [inline icicle-compiler/src/Icicle/Sea/Eval/Base.hs]: yeah you can maybe do something like this:
```hs
withCPiano' :: ForeignPiano -> (CPiano -> EitherT x IO a) -> EitherT x IO a
withCPiano' fp io =
  EitherT . withCPiano fp $ runEitherT . io
```

Comment (tranma) [inline icicle-compiler/src/Icicle/Sea/Eval/Base.hs]: I changed sfSnapshot was it was being liftIO'd anyway

**Comment by jystic** [comment]
👍

Just a heads up, blizzard will not have a `FilePath` but a `Conduit` connected to S3. So we'll need to change this a bit when we do that integration. I suspect we'll need to pass in the puller in order to make that work, not sure.

**Pull request merged**